pkg/osm/callbacks.go

```diff
@@ -24,7 +29,8 @@ import (
 func (cp *ControlPlane[T]) ProxyConnected(ctx context.Context, connectionID int64) error {
 	// When a new Envoy proxy connects, ValidateClient would ensure that it has a valid certificate,
 	// and the Subject CN is in the allowedCommonNames set.
-	certCommonName, certSerialNumber, err := utils.ValidateClient(ctx)
+	// If the current issuer is singing then we should use the SPIFFE ID instead of common name
```

Comment:
The spiffe id will only be in the certificates once the signing issuer is issuing all the certificates.  Once that is the case we can switch to using the SPIFFE ID to gather the metadata about the proxy connecting instead of the Common name. 

When using SPIFFE as the standard, the identity in the certificate is the URI SAN not the common name.

This will be required once we are using SPIRE since SPIRE will create the SPIFFE ID in the URI field instead of common name.

Reply:
Should move this comment to below as well as update the line 30/31

Reply:
updated comment